tests/_test_utils/test_coordinate.py

```diff
@@ -39,140 +39,127 @@
-        assert len(lons) == 3
-        assert all(lats == [0, 5, 10])
-
-        ref_lats = [0, 0]
-        ref_lons = [0, 10]
-        lats, lons = coordinate.latlon_points(ref_lats[0], ref_lons[0], ref_lats[1], ref_lons[1],
-                                              numpoints=3, connection="linear")
-        assert len(lats) == 3
-        assert len(lons) == 3
-        assert all(lons == [0, 5, 10])
-
-        lats, lons = coordinate.latlon_points(ref_lats[0], ref_lons[0], ref_lats[1], ref_lons[1],
-                                              numpoints=3, connection="linear")
-        assert len(lats) == 3
-        assert len(lons) == 3
-        assert all(lons == [0, 5, 10])
-
-    def test_greatcircle(self):
-        ref_lats = [0, 10]
-        ref_lons = [0, 0]
-
-        lats, lons = coordinate.latlon_points(ref_lats[0], ref_lons[0], ref_lats[1], ref_lons[1],
-                                              numpoints=2, connection="greatcircle")
-        assert len(lats) == len(ref_lats)
-        assert lats == ref_lats
-        assert len(lons) == len(ref_lons)
-        assert lons == ref_lons
+    """
+    Tests for generating lat/lon points along paths.
+    """
 
+    @pytest.mark.parametrize("ref_lats, ref_lons, numpoints, connection, expected_lats, expected_lons", [
+        ([0, 10], [0, 0], 2, "linear", [0, 10], [0, 0]),
+        ([0, 10], [0, 0], 3, "linear", [0, 5, 10], [0, 0, 0]),
+        ([0, 0], [0, 10], 3, "linear", [0, 0, 0], [0, 5, 10]),
+    ])
+    def test_linear(self, ref_lats, ref_lons, numpoints, connection, expected_lats, expected_lons):
+        """
+        Test generating linear lat/lon points.
+        """
         lats, lons = coordinate.latlon_points(ref_lats[0], ref_lons[0], ref_lats[1], ref_lons[1],
-                                              numpoints=3, connection="linear")
-        assert len(lats) == 3
-        assert len(lons) == 3
-        assert all(np.asarray(lats) == [0, 5, 10])
-
-        ref_lats = [0, 0]
-        ref_lons = [0, 10]
+                                              numpoints=numpoints, connection=connection)
+        np.testing.assert_array_equal(lats, expected_lats)
+        np.testing.assert_array_equal(lons, expected_lons)
+
+    @pytest.mark.parametrize("ref_lats, ref_lons, numpoints", [
+        ([0, 10], [0, 0], 2),
+        ([0, 10], [0, 0], 3),
+    ])
+    def test_greatcircle(self, ref_lats, ref_lons, numpoints):
+        """
+        Test generating lat/lon points along a great circle path.
+        """
         lats, lons = coordinate.latlon_points(ref_lats[0], ref_lons[0], ref_lats[1], ref_lons[1],
-                                              numpoints=3, connection="linear")
-        assert len(lats) == 3
-        assert len(lons) == 3
-        assert all(np.asarray(lons) == [0, 5, 10])
+                                              numpoints=numpoints, connection="greatcircle")
+        assert len(lats) == numpoints
+        assert len(lons) == numpoints
```

Comment:
We should also check for the expected value, like in assert all(np.asarray(lons) == [0, 5, 10])